sdk/security_keyvault/src/client.rs

```diff
@@ -137,13 +157,25 @@ impl<'a, T: TokenCredential> KeyClient<'a, T> {
             .await
             .unwrap();
 
-        let body = resp.text().await.unwrap();
-
-        let body_serialized = serde_json::from_str::<serde_json::Value>(&body).unwrap();
-
-        if let Some(err) = body_serialized.get("error") {
-            let msg = err.get("message").ok_or(Error::UnparsableError)?;
-            Err(Error::General(msg.to_string()))
+        let body = resp.text().await.with_context(ErrorKind::Io, || {
+            format!("failed to read response body text. uri:{}", uri)
+        })?;
+        let body_deserialized = serde_json::from_str::<serde_json::Value>(&body).unwrap();
+
+        if let Some(err) = body_deserialized.get("error") {
+            let msg = err.get("message").ok_or_else(|| {
+                Error::with_message(
+                    ErrorKind::DataConversion,
+                    format!(
```

Comment:
In #781, I change this to allow a `|| ` as well.

Reply:
Cool.  I can change this once #781 is merged.